### PR TITLE
[rack] use the Rack env instead of the Rack request object

### PR DESCRIPTION
**Pull request by palazzem**
### What it does

Uses directly the `env` object (we have one `env` per request) instead of the Rack `Request`. This is required because in some frameworks (like Rails) the `request.path_info` could be changed at runtime, especially during exception handling. This makes results not reliable for our `http.url` tag. For instance:
```ruby
# suppose we call a Rails endpoint like /error
request = ::Rack::Request.new(env)
assert_equal(request.path_info, '/error')
status, headers, response = @app.call(env)
assert_equal(request.path_info, '/error')  # this fails!
```

The stored value now is `/500`.

### CODE REVIEW

File: lib/ddtrace/contrib/rack/middlewares.rb [hunk 1]
```diff
@@ -55,15 +52,22 @@ def call(env)
           request_span.set_error(e)
           raise e
         ensure
+          # the source of truth in Rack is the PATH_INFO key that holds the
+          # URL for the current request; some framework may override that
+          # value, especially during exception handling and because of that
+          # we prefer using the `REQUEST_URI` if this is available.
```
Comment (ufoot) [inline lib/ddtrace/contrib/rack/middlewares.rb]: I'm OK with this, globally, but on a later release we might dig into it and figure out why a framework would do that, because maybe the `PATH_INFO`, when it's overridden, is overridden to a value that is technically less related to the "physical" http request but more meaningful in the app context.

Comment (palazzem) [inline lib/ddtrace/contrib/rack/middlewares.rb]: actually `PATH_INFO` must include the request URL as defined in Rack docs: http://www.rubydoc.info/github/rack/rack/file/SPEC#The_Environment

We may double check on Rails why it's happening. This is just a reasonable default because we know about this exception, but definitely something to take in consideration if it happens also for other frameworks.

Comment (ufoot) [inline lib/ddtrace/contrib/rack/middlewares.rb]: :+1: good to go, and anyway, this is orthogonal to / independent from the short-term goal we have: have rack support to avoid loosing traces.
